internal/internal_event_handlers.go

```diff
@@ -568,16 +578,36 @@ func (wc *workflowEnvironmentImpl) GetVersion(changeID string, minSupported, max
 		// GetVersion for changeID is called first time in replay mode, use DefaultVersion
 		version = DefaultVersion
 	} else {
-		// GetVersion for changeID is called first time (non-replay mode), we need to generate a marker decision for it.
+		// GetVersion for changeID is called first time (non-replay mode), generate a marker decision for it.
+		// Also upsert search attributes to enable ability to search by changeVersion.
 		version = maxSupported
 		wc.decisionsHelper.recordVersionMarker(changeID, version, wc.GetDataConverter())
+		wc.UpsertSearchAttributes(createSearchAttributesForChangeVersion(changeID, version, wc.changeVersions))
```

Comment:
Why are we upserting all the versions instead of the new one?

Reply:
Because value of search attribute can only be override, we cannot append data to array type.  
(array type is treated same as other basic type)